Hivemind/Assets/Scripts/Gathering.cs

```diff
@@ -59,23 +105,47 @@ private void Update()
         {
             case State.Idle:
                 agent.isStopped = true;
-                target = findResource();
-                if (target != null)
-                {
-                    agent.isStopped = false;
-                    nextHarvest = target.DecreaseFutureResources(CarryAmount - inventoryAmount);
-                    agent.SetDestination(target.GetPosition());
-                    state = State.MovingToResource;
-                }
+                TargetResource();
                 break;
+
             case State.Scouting:
+                if (!preparingReturn)
```

Comment:
hoezo not? is misschien hier in deze constructie 1 if else handiger? met daarbij ofc 1 bool ipv 2

Reply:
Dit is voor het starten van de coroutine om terug te keren na niks vinden, de if statement hieronder is voor het lopen naar een niewe plek om te scouten. Deze hebben geen relatie met elkaar behalve dat ze beide bij het scout behavior horen. Else if is hier dus niet van toepassing